bootstrap/090_post_setup.sql

```diff
@@ -329,22 +330,39 @@ BEGIN
                 total_inserted_rows := total_inserted_rows + new_inserted_rows;
             end for;
             let new_running timestamp := (select max(end_time) from internal_reporting_mv.warehouse_load_history where warehouse_name = :wh_name);
-            insert into INTERNAL.TASK_WAREHOUSE_LOAD_EVENTS SELECT :dt, true, :wh_name, :input, OBJECT_CONSTRUCT('oldest_running', :new_running, 'new_records', coalesce(:total_inserted_rows, 0))::VARIANT;
+            insert into INTERNAL.TASK_WAREHOUSE_LOAD_EVENTS SELECT :dt, true, :wh_name, :input, OBJECT_CONSTRUCT('oldest_running', :new_running, 'new_records', coalesce(:total_inserted_rows, 0), 'task_run_id', :task_run_id, 'end', current_timestamp())::VARIANT;
         exception
             when other then
                 SYSTEM$LOG_ERROR(OBJECT_CONSTRUCT('error', 'Exception occurred while refreshing ' || :wh_name || ' events.', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate));
-                insert into INTERNAL.TASK_WAREHOUSE_LOAD_EVENTS SELECT :dt, false, :wh_name, :input, OBJECT_CONSTRUCT('Error type', 'Other error', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate)::variant;
+                insert into INTERNAL.TASK_WAREHOUSE_LOAD_EVENTS SELECT :dt, false, :wh_name, :input, OBJECT_CONSTRUCT('Error type', 'Other error', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate, 'task_run_id', :task_run_id, 'end', current_timestamp())::variant;
                 RAISE;
         end;
     end for;
 
 exception
     when other then
         SYSTEM$LOG_ERROR(OBJECT_CONSTRUCT('error', 'Exception occurred while refreshing all warehouse events.', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate));
-        insert into INTERNAL.TASK_WAREHOUSE_LOAD_EVENTS SELECT :dt, false, 'all', null, OBJECT_CONSTRUCT('Error type', 'Other error', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate)::variant;
+        insert into INTERNAL.TASK_WAREHOUSE_LOAD_EVENTS SELECT :dt, false, 'all', null, OBJECT_CONSTRUCT('Error type', 'Other error', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate, 'task_run_id', :task_run_id, 'end', current_timestamp())::variant;
         RAISE;
 end;
 
+-- The admin.materialization_status view has to be created from a task because it reads from the TASK_HISTORY UDTF.
+CREATE OR REPLACE TASK TASKS.CREATE_MATERIALIZATION_STATUS
+    ALLOW_OVERLAPPING_EXECUTION = FALSE
+    USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "X-Small"
+    AS
+DECLARE
+  sql STRING;
+BEGIN
+    call internal.create_materialization_status() into :sql;
+    execute immediate sql;
```

Comment:
Have to play this trick (like for probe_monitoring), else creating a view or UDTF that interacts with the snowflake.information_schema UDTFs will fail.